private_set_intersection/javascript/bindings/client.cpp

```diff
@@ -34,56 +36,105 @@ EMSCRIPTEN_BINDINGS(PSI_Client) {
-                  // We need to convert to a JS array explicitly because JS
-                  // doesn't know about vector<int64_t>.
-                  StatusOr<emscripten::val> result;
-                  auto status =
-                      self.GetIntersection(server_setup, server_response);
+                  StatusOr<psi_proto::Request> request;
+                  auto status = self.CreateRequest(string_vector);
                   if (status.ok()) {
-                    // Convert int64_t to int32_t for JS
-                    const std::vector<std::int64_t> unsupported_result =
-                        status.ValueOrDie();
-                    const std::vector<std::int32_t> supported_result(
-                        unsupported_result.begin(), unsupported_result.end());
-                    // Convert vector to JS array
-                    emscripten::val array = emscripten::val::array(
-                        supported_result.begin(), supported_result.end());
-                    result = std::move(StatusOr<emscripten::val>(array));
+                    request = status.ValueOrDie();
                   } else {
-                    result = status.status();
+                    request = status.status();
                   }
-                  return ToJSObject(result);
-                }))
-      .function("GetIntersectionSize",
-                optional_override([](const PsiClient& self,
-                                     const std::string& server_setup,
-                                     const std::string& server_response) {
-                  // We need to convert to an int32 explicitly because JS
-                  // doesn't have 64-bit integers.
-                  StatusOr<int32_t> result;
-                  auto status =
-                      self.GetIntersectionSize(server_setup, server_response);
-                  if (status.ok()) {
-                    result = status.ValueOrDie();
-                  } else {
-                    result = status.status();
-                  }
-                  return ToJSObject(result);
+                  return ToSerializedJSObject(request);
                 }))
+      .function(
+          "GetIntersection",
+          optional_override([](const PsiClient& self,
+                               const emscripten::val& server_setup_array,
+                               const emscripten::val& server_response_array) {
+            const std::size_t server_setup_length =
+                server_setup_array["length"].as<std::size_t>();
+            std::string server_setup_string(server_setup_length, '\0');
+            for (std::size_t i = 0; i < server_setup_length; i++) {
+              server_setup_string[i] = server_setup_array[i].as<std::uint8_t>();
+            }
+
+            const std::size_t server_response_length =
+                server_response_array["length"].as<std::size_t>();
+            std::string server_response_string(server_response_length, '\0');
```

Comment:
If it was possible to get a pointer to the raw Uint8Array data from emscripten::val directly, we could use ParseFromArray below and avoid copying to a string. But I'm not sure if emscripten allows this.

Reply:
I tried to use the array iterators provided by emscripten::val::array, but was not successful in getting it to work properly using `ParseFromArray`. 

There might a way to get a pointer to `val`, I would prefer to release what is working now and work on this optimization in the future.

Reply:
Sounds good to me.